backend/cn/lib/core_to_mucore.ml

```diff
@@ -1527,8 +1527,16 @@ let normalise_file ~inherit_loc ((fin_markers_env : CAE.fin_markers_env), ail_pr
   let fin_d_st = CAE.{ inner = Pmap.find fin_marker markers_env; markers_env } in
   let env = C.init_env tagDefs (fetch_enum fin_d_st) (fetch_typedef fin_d_st) in
   let@ env = C.add_datatype_infos env ail_prog.cn_datatypes in
-  let@ env = C.register_cn_functions env ail_prog.cn_functions in
-  let@ lfuns = ListM.mapM (C.translate_cn_function env) ail_prog.cn_functions in
+  let cn_funs = Builtins.builtin_cn_funs @ ail_prog.cn_functions in
+  let@ env = C.register_cn_functions env cn_funs in
+  let@ lfuns =
+    ListM.mapiM
+      (fun index ->
+        C.translate_cn_function
+          ~builtin:(index < List.length Builtins.builtin_cn_funs)
```

Comment:
`List.length` is a linear time function called on each iteration here. Perhaps simpler to just have two calls to translate_cn_function, one for the builtins, and another for ail_prog.cn_functions?

Reply:
Right yeah I forgot `length` works like this in OCaml